### PR TITLE
Update 3.9.0

### PR DESCRIPTION
**Pull request by dalcinl**

### CODE REVIEW

**Comment by dalcinl** [comment]
@joseeroman Welcome to the team! I a couple days you will be given write access to this repo (a conda-forge bot will do it automatically).

**Comment by dalcinl** [comment]
@joseeroman Please note the last lines in `recipe/build.sh`. After make install, I'm removing `*.log` and `$PREFIX/lib/slepc/conf/files`. I don't think you really meant them installed. Please review and eventually fix upstream SLEPc.  I'm also removing `examples` and `datafiles` from the conda package, I have not objections from SLEPc installing them,  I just don't see the point of having this stuff in the conda-forge binary package.

**Comment by dalcinl** [comment]
@minrk This is ready to go.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -38,7 +37,7 @@ requirements:
     - libgfortran  # [osx]
     - blas 1.* {{blas}}
     {% if blas == 'openblas' %}
-    - openblas 0.2.20|0.2.20.*
+    - openblas 0.2.20
```
Comment (minrk) [inline recipe/meta.yaml]: Why is the openblas pin updated here?

**Comment by dalcinl** [comment]
I run `conda-build` (v2.1.18) locally and got a complaint about the version spec. I pushed the same change in the PETSc 3.9 update PR. I really don't know what's going on with my conda-build tool, I have to pass --no-verify otherwise it even complains about the `mpich >= 3.2` spec. The hypre package is also using plain `openblas 0.2.20`. However numpy uses `0.2.20|0.2.20.*`. I'm not sure what to do here, should I revert the blas pin to the original form?

**Comment by joseeroman** [comment]
@dalcinl `make install` in SLEPc no longer copies the log files
https://bitbucket.org/slepc/slepc/commits/8b2fdfc

File: recipe/meta.yaml (hunk 1 above)
Comment (dalcinl) [reply recipe/meta.yaml]: Sorry, I didn't reply here but in the PR. So, what should I do here? Do you want me to revert this change?

Comment (minrk) [reply recipe/meta.yaml]: I think reverting this is the right thing to do. Only change it if it's fixing something.

**Comment by dalcinl** [comment]
@joseeroman Look at my last commit. I'm removing the debug stuff generated in macOS. PETSc 3.9 does not install that. Just FYI, I'll not complain if you decide to keep it for SLEPc's `make install`, but maybe we did not really intended to do that, so now you know for sure.

**Pull request merged**